presto-tests/src/test/java/io/prestosql/execution/TestFlushingStageState.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.execution;
+
+import com.google.common.collect.ImmutableMap;
+import io.prestosql.spi.QueryId;
+import io.prestosql.testing.DistributedQueryRunner;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.Test;
+
+import static io.prestosql.SessionTestUtils.TEST_SESSION;
+import static io.prestosql.execution.QueryState.RUNNING;
+import static io.prestosql.execution.StageState.CANCELED;
+import static io.prestosql.execution.StageState.FLUSHING;
+import static io.prestosql.execution.TestQueryRunnerUtil.createQuery;
+import static io.prestosql.execution.TestQueryRunnerUtil.createQueryRunner;
+import static io.prestosql.execution.TestQueryRunnerUtil.waitForQueryState;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static org.testng.Assert.assertEquals;
+
+public class TestFlushingStageState
+{
+    private DistributedQueryRunner queryRunner;
+
+    @Test(timeOut = 30_000)
+    public void testFlushingState()
+            throws Exception
+    {
+        queryRunner = createQueryRunner(ImmutableMap.of("node-scheduler.include-coordinator", "false"));
+        QueryId queryId = createQuery(queryRunner, TEST_SESSION, "SELECT * FROM tpch.sf1000.lineitem limit 1");
+        waitForQueryState(queryRunner, queryId, RUNNING);
+
+        QueryInfo queryInfo = queryRunner.getCoordinator().getFullQueryInfo(queryId);
+        while (queryInfo.getOutputStage().get().getState() != FLUSHING) {
```

Comment:
please use `assertEventually`

Reply:
fixed - also moved the creation of queryRunner to an unbounded `setup`. I think the test after setup should run within 30sec.